tssc/step_implementers/push_artifacts/maven.py

```diff
@@ -34,18 +32,20 @@ def _validate_step_config(self, step_config):
             Step configuration to validate.
         """
         if 'url' not in step_config or not step_config['url']:
-            raise ValueError('Key (url) must have none empty value in the step configuration')
-        if 'user' not in step_config or not step_config['user']:
-            raise ValueError('Key (user) must have none empty value in the step configuration')
-        if 'password' not in step_config or not step_config['password']:
-            raise ValueError('Key (password) must have none empty value in the step configuration')
+            raise ValueError('url must have none empty value in the step configuration')
 
     def _run_step(self, runtime_step_config):
 
-        # ----- get config items
         url = runtime_step_config['url']
-        user = runtime_step_config['user']
-        password = runtime_step_config['password']
+
+        if 'user' not in runtime_step_config or not runtime_step_config['user']:
+            raise ValueError('user must have none empty value in the run configuration')
```

Comment:
@ppremru these shouldn't be required. repos can have anonymous push. and or push to a user without a password needed.

The only error you should throw in params checking is if a password is specified without a user name. otherwise have to throw error at runtime if the repo you are pushing to rejects your crednetials (or lack there of)

Reply:
+1

Reply:
Made changes to support optional user  and password.